### PR TITLE
chore: add unit tests to sdk connect handlers

### PR DESCRIPTION
**Pull request by omridan159**
## **Description**

 Add unit tests to SDKConnect handlers to make the code more stable and reliable
 
## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/core/SDKConnect/handlers/checkPermissions.test.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+jest.mock('../Connection', () => ({
+  RPC_METHODS: jest.requireActual('../Connection').RPC_METHODS,
+}));
+jest.mock('../../Engine');
+jest.mock('@metamask/preferences-controller');
+jest.mock('@metamask/approval-controller');
+jest.mock('../utils/DevLogger');
+
+describe('checkPermissions', () => {
+  let connection = {} as unknown as Connection;
```
Comment (abretonc7s) [inline app/core/SDKConnect/handlers/checkPermissions.test.ts:16]: probably should use corrct types here

Comment (abretonc7s) [reply app/core/SDKConnect/handlers/checkPermissions.test.ts:16]: Just implement as a partial for what is needed for the unit test:
```
let connection: Partial<Connection> = ...
```

File: app/core/SDKConnect/handlers/checkPermissions.test.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  let engine = {
+    context: {},
+  } as unknown as typeof Engine;
```
Comment (abretonc7s) [inline app/core/SDKConnect/handlers/checkPermissions.test.ts:19]: Did you check on the test files in the codebase, they may have already implemented Engine?

File: app/core/SDKConnect/handlers/handleConnectionReady.test.ts [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+
+    originatorInfo = {
+      apiVersion: '0.2.0',
+      source: 'fakeExtensionId',
+      url: 'fakeUrl',
+      icon: 'fakeIcon',
+      color: 'fakeColor',
+      title: 'asdfsdf',
+      platform: 'fakePlatform',
+    } as OriginatorInfo;
```
Comment (abretonc7s) [inline app/core/SDKConnect/handlers/handleConnectionReady.test.ts:50]: let's avoid the as Casting and try to use real types

**Review by abretonc7s — DISMISSED** [review]
Didn't go over all the tests but overall would need to change the cast using as with the using real types in the variables declarations

File: app/core/SDKConnect/handlers/checkPermissions.test.ts (hunk 1 above)
Comment (omridan159) [reply app/core/SDKConnect/handlers/checkPermissions.test.ts:16]: We can't do that cause then we get:
Type 'Partial<Connection>' is not assignable to type 'Connection'.
When we try to pass it to the `checkPermissions` params...

File: app/core/SDKConnect/handlers/checkPermissions.test.ts (hunk 2 above)
Comment (omridan159) [reply app/core/SDKConnect/handlers/checkPermissions.test.ts:19]: I have checked now, and I didn't saw something...

**Review by deeeed — APPROVED** [review]

**Review by andreahaku — APPROVED** [review]
LGTM

**Pull request merged**